docs: add document for dns chaos (#3)
chaos-mesh/website: docs/chaos_experiments/dns_chaos.md

---
id: dnschaos_experiment
title: DNSChaos Experiment
sidebar_label: DNSChaos Experiment
---

> This document describes how to create DNSChaos experiments in Chaos Mesh.

Reviewer: Please explain DNS chaos. What does it do? What's the use scenario

Author: add in https://github.com/chaos-mesh/website/pull/3/commits/904d01364304c0baaf0e3f625c8a6bdf5a02f075

---
id: dnschaos_experiment
title: DNSChaos Experiment
sidebar_label: DNSChaos Experiment
---

This document describes how to create DNSChaos experiments in Chaos Mesh.

> DNSChaos allows you to simulate DNS responses in fault situations including DNS errors or random IP addresses.

Reviewer: ```suggestion
DNSChaos allows you to simulate fault DNS responses such as a DNS error or a random IP address after a request is sent.
```

## Deploy DNS service for chaos

To create DNSChaos experiments in Chaos Mesh, you need to deploy a DNS service in Chaos Mesh by executing the command below:

```bash
helm upgrade chaos-mesh helm/chaos-mesh --namespace=chaos-testing --set dnsServer.create=true
```

When the deployment finishes, check the status of this DNS service:

```bash
kubectl get pods -n chaos-testing -l app.kubernetes.io/component=chaos-dns-server
```

Make sure the Pod's `STATUS` is `Running`.

## Configuration file

Below is a sample DNSChaos configuration file:

```yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: DNSChaos
metadata:
  name: busybox-dns-chaos
spec:
  action: random
  scope: inner
  mode: all
  selector:
    namespaces:
      - busybox
  duration: "90s"
  scheduler:
    cron: "@every 100s"
```

For more sample files, see [examples](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples). You can edit them as needed.

## Fields description

- **action**: Defines the chaos action for DNS chaos. Supported actions are:
    - `error` - Get an error when sending the DNS request
    - `random` - Get a random IP when sending the DNS request

> * **scope** Defines the scope of the DNS chaos. Supported scopes are:

Reviewer: ```suggestion
* **scope**: Defines the scope of the DNS chaos. Supported scopes are:
```

    - `outer` - DNS chaos only works on outer hosts of the Kubernetes cluster
    - `inner`- DNS chaos only works on inner hosts of the Kubernetes cluster
    - `all` - DNS chaos works on all hosts.

> * **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).

Reviewer: ```suggestion
* **selector** Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```

> * **selector** Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).

Reviewer: ```suggestion
* **selector**: Specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).
```

## Notes

- Currently, DNSChaos only supports record types `A` and `AAAA`.

> - The Chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin, If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with the K8s CoreDNS service:

Reviewer: ```suggestion
- The chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin. If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with that of the K8s CoreDNS service as shown below:
```

---
id: dnschaos_experiment
title: DNSChaos Experiment
sidebar_label: DNSChaos Experiment
---

This document describes how to create DNSChaos experiments in Chaos Mesh.

DNSChaos allows you to simulate response for DNS requests, include errors or random IP addresses.

## Deploy DNS service for chaos

> To create DNSChaos experiments in Chaos Mesh, need to deploy a DNS service in Chaos Mesh by executing the command below:

Reviewer: Please add the subjective before "need"

```

Then check the status of this DNS service:

```bash
kubectl get pods -n chaos-testing -l app.kubernetes.io/component=chaos-dns-server
```

Make sure the pod's `STATUS` is `Running`.

## Configuration file

Below is a sample DNSChaos configuration file:

```yaml
apiVersion: chaos-mesh.org/v1alpha1
kind: DNSChaos
metadata:
  name: busybox-dns-chaos
spec:
  action: random
  scope: inner
  mode: all
  selector:
    namespaces:
      - busybox
  duration: "90s"
  scheduler:
    cron: "@every 100s"
```

For more sample files, see [examples](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples). You can edit them as needed.

## Fields description

* **action** Defines the chaos action about DNS. Supported action: `error` / `random`.
    - `error` Get error when send DNS request.
    - `random` Get random IP when send DNS request.

* **scope** Defines the scope in which the DNS chaos works. Supported scope: `outer` / `inner` / `all`.
    - `outer` DNS chaos only works on the host outer of the Kubernetes cluster.
    - `inner` DNS chaos only works on the inner host in the Kubernetes cluster.
    - `all` DNS chaos works on all hosts.

> * **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).

Reviewer: - **action**: Defines the chaos action for DNS chaos. Supported actions are:
    - `error` - Get an error when sending the DNS request
    - `random` - Get a random IP when sending the DNS request

* **scope** Defines the scope of the DNS chaos. Supported scopes are:
    - `outer` - DNS chaos only works on outer hosts of the Kubernetes cluster
    - `inner`- DNS chaos only works on inner hosts of the Kubernetes cluster
    - `all` - DNS chaos works on all hosts.

* **selector** specifies the target pods for chaos injection. For more details, see [Define the Scope of Chaos Experiment](../user_guides/experiment_scope.md).

## Notes

> - Now DNS chaos only supports record type `A` and `AAAA`.

Reviewer: ```suggestion
- Currently, DNSChaos only supports record types `A` and `AAAA`.
```

> - Chaos DNS Service runs coredns with plugin [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos), If the coredns service in your K8s cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of chaos DNS service consistent with the K8s coredns service:

Reviewer: ```suggestion
- The Chaos DNS service runs CoreDNS with the  [k8s_dns_chaos](https://github.com/chaos-mesh/k8s_dns_chaos) plugin, If the CoreDNS service in your Kubernetes cluster contains some special configurations, you can edit configMap `dns-server-config` to make the configuration of the chaos DNS service consistent with the K8s CoreDNS service:
```
